src/librustc/diagnostics.rs

```diff
@@ -791,8 +791,7 @@ trait Foo {
     fn bar(&self);
 }
 
-// we now declare a function which takes an object with Foo trait implemented
-// as parameter
+// we now declare a function which takes an object implementing the Foo trait
```

Comment:
I don't really agree; but I do agree that this is unambiguously an improvement over the old wording.